server/game/cards/09.3-JfS/IsawaHeiko.js

```diff
@@ -0,0 +1,30 @@
+const DrawCard = require('../../drawcard.js');
+const { CardTypes, Elements } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl');
+
+class IsawaHeiko extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Switch a character\'s M and P skill',
+            when: {
+                onCardPlayed: (event, context) => {
+                    return event.card.hasTrait(Elements.Water) &&
+                        context.game.isDuringConflict() &&
+                        event.player === context.player;
+                }
+            },
+            target: {
+                cardType: CardTypes.Character,
+                cardCondition: card => !card.hasDash() && card.isParticipating(),
+                gameAction: AbilityDsl.actions.cardLastingEffect({
+                    effect: AbilityDsl.effects.switchBaseSkills()
+                })
```

Comment:
It looks as though Heiko effect is persistent without an expiration.  Please add `duration: Durations.persistent`

Reply:
The actual card has expiration (end of phase), just a miss by 5rdb.

Reply:
Yep, just been informed.  Either way this still needs a change, this now needs to be `duration: Durations.UntilEndOfPhase`